cpp/src/arrow/engine/substrait/util.cc

```diff
@@ -107,19 +111,36 @@ class SubstraitExecutor {
   bool plan_started_;
   compute::ExecContext exec_context_;
   std::shared_ptr<SubstraitSinkConsumer> sink_consumer_;
+  const ConversionOptions& conversion_options_;
 };
 
 }  // namespace
 
 Result<std::shared_ptr<RecordBatchReader>> ExecuteSerializedPlan(
-    const Buffer& substrait_buffer, const ExtensionIdRegistry* extid_registry,
-    compute::FunctionRegistry* func_registry) {
+    const Buffer& substrait_buffer, const PythonTableProvider& table_provider,
+    const ExtensionIdRegistry* registry, compute::FunctionRegistry* func_registry) {
```

Comment:
```suggestion
    const Buffer& substrait_buffer,
    const ExtensionIdRegistry* registry, compute::FunctionRegistry* func_registry, const ConversionOptions& conversion_options = {}) {
```

I suspect it will only be a matter of time before we want to specify other conversion options.  Also, by defaulting to `{}` we can avoid changing existing tests that are unrelated to the table provider.

Reply:
One thing, if we do this, I think we need to expose `Declaration` to Cython. Because we have to define the `NamedTableProvider` and bind it to the options. Is that okay?

Reply:
Please ignore my previous comment. It is already exposed in `_exec_plan.pyx`